plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/SecurityParams.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg.catalog.rest;
+
+import com.google.common.collect.ImmutableMap;
+
+public interface SecurityParams
+{
+    void configure(ImmutableMap.Builder<String, String> properties);
```

Comment:
Personal preference, feel free to ignore, but I'd rather have this be something like
```suggestion
    Map<String, String> requiredSecurityProperties();
```

rather than having side effects on the Map builder.

Reply:
I went ahead and used this approach and just stored the properties as opposed to the config object.